movebucket_test.go

```diff
@@ -0,0 +1,482 @@
+					dstBucket = retrieveChildBucket(t, tx, tc.dstBucketPath...)
+				}
+
+				if tc.expActErr == nil {
+					bucketToMove := srcBucket.Bucket([]byte(tc.bucketToMove))
+					wErr := dumpBucket([]byte(tc.bucketToMove), bucketToMove, dumpBucketBeforeMoving)
+					if wErr != nil {
+						t.Fatalf("error dumping bucket %v to file %v", bucketToMove.String(), dumpBucketBeforeMoving)
+					}
+				}
+
+				mErr := srcBucket.MoveBucket([]byte(tc.bucketToMove), dstBucket)
+				require.ErrorIs(t, mErr, tc.expActErr)
+
+				return nil
+			}); err != nil {
+				t.Fatal(err)
+			}
+			db.MustCheck()
+
+			// skip assertion if failure expected
+			if tc.expActErr != nil {
+				return
+			}
+
+			// assert
+			if err := db.View(func(tx *bbolt.Tx) error {
+
+				var movedBucket *bbolt.Bucket
+				srcBucket := retrieveParentBucket(t, tx, tc.srcBucketPath...)
+
+				if movedBucket = srcBucket.Bucket([]byte(tc.bucketToMove)); movedBucket != nil {
+					t.Fatalf("expected childBucket %v to be moved from srcBucket %v", tc.bucketToMove, srcBucket)
+				}
+
+				dstBucket := retrieveParentBucket(t, tx, tc.dstBucketPath...)
+				if dstBucket == nil {
+					if movedBucket = tx.Bucket([]byte(tc.bucketToMove)); movedBucket == nil {
+						t.Fatalf("expected childBucket %v to be child of dstBucket %v", tc.bucketToMove, "root bucket")
+					}
+				} else {
+					if movedBucket = dstBucket.Bucket([]byte(tc.bucketToMove)); movedBucket == nil {
+						t.Fatalf("expected childBucket %v to be child of dstBucket %v", tc.bucketToMove, dstBucket)
+					}
+				}
+
+				wErr := dumpBucket([]byte(tc.bucketToMove), movedBucket, dumpBucketAfterMoving)
+				if wErr != nil {
+					t.Fatalf("error dumping bucket %v to file %v", movedBucket.String(), dumpBucketAfterMoving)
+				}
+
+				beforeBucket := readBucketFromFile(t, dumpBucketBeforeMoving)
+				afterBucket := readBucketFromFile(t, dumpBucketAfterMoving)
+
+				if !bytes.Equal(beforeBucket, afterBucket) {
+					t.Fatalf("bucket's content before moving is different than after moving")
+				}
+
+				return nil
+			}); err != nil {
+				t.Fatal(err)
+			}
+			db.MustCheck()
+
+			t.Cleanup(func() {
+				os.RemoveAll(dumpBucketBeforeMoving)
+				os.RemoveAll(dumpBucketAfterMoving)
+			})
+		})
+	}
+}
+
+func createBucketIfNotExist(t testing.TB, tx *bbolt.Tx, paths ...string) *bbolt.Bucket {
+	t.Helper()
+
+	var bk *bbolt.Bucket
+	var err error
+
+	// skip if root bucket (i.e. bucket name == "")
+	if paths[0] != "" {
+		bk, err = tx.CreateBucketIfNotExists([]byte(paths[0]))
+		if err != nil {
+			t.Fatalf("error creating bucket %v: %v", paths[0], err)
+		}
+		insertRandKeysValuesBucket(t, bk, rand.Intn(4096))
+	}
+
+	for _, key := range paths[1:] {
+		if bk == nil {
+			bk, err = tx.CreateBucketIfNotExists([]byte(key))
+		} else {
+			bk, err = bk.CreateBucketIfNotExists([]byte(key))
+		}
+		if err != nil {
+			t.Fatalf("error creating bucket %v: %v", key, err)
+		}
+		insertRandKeysValuesBucket(t, bk, rand.Intn(4096))
+	}
```

Comment:
```suggestion
	// skip if root bucket (i.e. bucket name == "")
	for _, key := range paths {
		if bk == nil || len(key) == 0 {
			bk, err = tx.CreateBucketIfNotExists([]byte(key))
		} else {
			bk, err = bk.CreateBucketIfNotExists([]byte(key))
		}
		if err != nil {
			t.Fatalf("error creating bucket %v: %v", key, err)
		}
		insertRandKeysValuesBucket(t, bk, rand.Intn(4096))
	}
```

Reply:
done

Reply:
cant create a bucket when `len(key)` == 0 

i adapted the code 👍🏽